api/src/main/java/ai/djl/ndarray/NDArrayAdapter.java

```diff
@@ -212,6 +212,16 @@ public NDArray get(NDIndex index) {
     /** {@inheritDoc} */
     @Override
     public void set(Buffer data) {
+        DataType arrayType = getDataType();
```

Comment:
I don't think we check list this:
1. We should always allow ByteBuffer to set for any DataType
2. Multiple dataType maps to the same Buffer type, we have to handle them differently

Reply:
I find that in the implementation of set(Buffer data) in different engines, like MXNet, PyTorch, TensorFlow, the buffer is already converted to ByteBuffer before being fed into the engines. Do you mean we should add this to NDArrayAdapter?

Reply:
By check listing those, is it out of the purpose of automatically converting the input datatype into the target array datatype?